src/training.rs

```diff
@@ -348,7 +368,29 @@ mod tests {
             AdamConfig::new(),
         );
 
-        let _model_trained =
-            train::<NdArrayAutodiffBackend>(trainset, &config, device, None).unwrap();
+        let mut weights_sets: Vec<Vec<f32>> = Vec::new();
+
+        for i in 0..n_splits {
+            let trainset = trainsets
+                .par_iter()
+                .enumerate()
+                .filter(|&(j, _)| j != i)
+                .flat_map(|(_, trainset)| trainset.clone())
+                .collect();
+            let model = train::<NdArrayAutodiffBackend>(trainset, &config, device, None);
+            weights_sets.push(model.unwrap().w.val().to_data().convert().value)
+        }
```

Comment:
This is roughly twice as fast:

```rust
        let weights_sets: Vec<Vec<f32>> = (0..n_splits).into_par_iter().map(|i| {
            let trainset = trainsets
                .iter()
                .enumerate()
                .filter(|&(j, _)| j != i)
                .flat_map(|(_, trainset)| trainset.clone())
                .collect();
            let model = train::<NdArrayAutodiffBackend>(trainset, &config, device, None);
            model.unwrap().w.val().to_data().convert().value
        }).collect();
```

It's still about 2x slower than the original implementation before this PR though, because there's now 4x as much work to be done, as the trainsets are being added together. Would it make sense to only use one of the trainsets instead?

```rust
        let weights_sets: Vec<Vec<f32>> = (0..n_splits).into_par_iter().map(|i| {
            let trainset = trainsets[i].clone();
            let model = train::<NdArrayAutodiffBackend>(trainset, &config, device, None);
            model.unwrap().w.val().to_data().convert().value
        }).collect();
```